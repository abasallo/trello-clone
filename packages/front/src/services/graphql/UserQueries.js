import gql from 'graphql-tag'

export const GET_USER = gql`
  query ($token: String, $email: String) {
    getUser(token: $token, email: $email) {
      id
      names
      surnames
      email
    }
  }
`

export const ADD_USER = gql`
  mutation ($token: String, $names: String, $surnames: String, $email: String!) {
    addUser(token: $token, user: { names: $names, surnames: $surnames, email: $email }) {
      id
      names
      surnames
      email
    }
  }
`
