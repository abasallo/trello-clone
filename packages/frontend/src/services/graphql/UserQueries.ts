import gql from 'graphql-tag'

export const GET_BOARDS = gql`
  query {
    getBoards {
      id
      name
    }
  }
`

export const ADD_BOARD = gql`
  mutation ($board: BoardInput!) {
    addBoard(board: $board) {
      id
      name
    }
  }
`

export const UPDATE_BOARD = gql`
  mutation ($board: BoardInput!) {
    updateBoard(board: $board) {
      id
      name
    }
  }
`

export const DELETE_BOARD = gql`
  mutation ($id: ID!) {
    deleteBoard(id: $id)
  }
`
