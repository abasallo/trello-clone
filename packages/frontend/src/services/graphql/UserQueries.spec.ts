import { ADD_BOARD, GET_BOARDS } from './UserQueries'

test('User add query must be initialized properly', () => expect(GET_BOARDS).toMatchSnapshot())

test('Password change query must be initialized properly', () => expect(ADD_BOARD).toMatchSnapshot())
