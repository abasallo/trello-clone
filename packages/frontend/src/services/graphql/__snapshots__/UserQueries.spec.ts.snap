// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password change query must be initialized properly 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "board",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "board",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "addBoard",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "BoardInput",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "board",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 96,
    "start": 0,
  },
}
`;

exports[`User add query must be initialized properly 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "getBoards",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 57,
    "start": 0,
  },
}
`;
